controllers/managedcluster.go

```diff
@@ -98,11 +111,25 @@ func (r *MultiClusterHubReconciler) ensureKlusterletAddonConfig(m *operatorsv1.M
 
 	utils.AddInstallerLabel(klusterletaddonconfig, m.GetName(), m.GetNamespace())
 
+	newSpec, _, err := unstructured.NestedMap(newKlusterletaddonconfig.Object, "spec")
+	if err != nil {
+		r.Log.Error(err, "Failed to fetch the spec from newKlusterletaddonconfig")
+		return ctrl.Result{}, err
+	}
+
+	err = unstructured.SetNestedMap(klusterletaddonconfig.Object, newSpec, "spec")
+	if err != nil {
+		r.Log.Error(err, "Failed to set the spec")
+		return ctrl.Result{}, err
+	}
+
 	err = r.Client.Update(ctx, klusterletaddonconfig)
```

Comment:
We need to only call this any time the installer label or spec changed. Otherwise, this would get called every reconcile.

Reply:
You mean whole ensureKlusterletAddonConfig function?

Reply:
Right before the `Update` call, just return early if the labels from `AddInstallerLabel` are correct and the spec is the same.